notebooks/00_quick_start/sar_movielens_with_azureml.ipynb

```diff
@@ -0,0 +1,668 @@
+    "    remove_seen=True, similarity_type=\"jaccard\", \n",
+    "    time_decay_coefficient=30, time_now=None, timedecay_formula=True, **header\n",
+    ")\n",
+    "\n",
+    "# train the SAR model\n",
+    "start_time = time.time()\n",
+    "\n",
+    "model.fit(train)\n",
+    "\n",
+    "train_time = time.time() - start_time\n",
+    "run.log(name=\"Training time\", value=train_time)\n",
+    "\n",
+    "start_time = time.time()\n",
+    "\n",
+    "top_k = model.recommend_k_items(test)\n",
+    "\n",
+    "test_time = time.time() - start_time\n",
+    "run.log(name=\"Prediction time\", value=test_time)\n",
+    "\n",
+    "# TODO: remove this call when the model returns same type as input\n",
+    "top_k['UserId'] = pd.to_numeric(top_k['UserId'])\n",
+    "top_k['MovieId'] = pd.to_numeric(top_k['MovieId'])\n",
+    "\n",
+    "# evaluate\n",
+    "eval_map = map_at_k(test, top_k, col_user=\"UserId\", col_item=\"MovieId\", \n",
+    "                    col_rating=\"Rating\", col_prediction=\"prediction\", \n",
+    "                    relevancy_method=\"top_k\", k=args.top_k)\n",
+    "eval_ndcg = ndcg_at_k(test, top_k, col_user=\"UserId\", col_item=\"MovieId\", \n",
+    "                      col_rating=\"Rating\", col_prediction=\"prediction\", \n",
+    "                      relevancy_method=\"top_k\", k=args.top_k)\n",
+    "eval_precision = precision_at_k(test, top_k, col_user=\"UserId\", col_item=\"MovieId\", \n",
+    "                                col_rating=\"Rating\", col_prediction=\"prediction\", \n",
+    "                                relevancy_method=\"top_k\", k=args.top_k)\n",
+    "eval_recall = recall_at_k(test, top_k, col_user=\"UserId\", col_item=\"MovieId\", \n",
+    "                          col_rating=\"Rating\", col_prediction=\"prediction\", \n",
+    "                          relevancy_method=\"top_k\", k=args.top_k)\n",
+    "\n",
+    "run.log(\"map\", eval_map)\n",
+    "run.log(\"ndcg\", eval_ndcg)\n",
+    "run.log(\"precision\", eval_precision)\n",
+    "run.log(\"recall\", eval_recall)\n",
+    "\n",
+    "os.makedirs('outputs', exist_ok=True)\n",
+    "os.makedirs('tmp', exist_ok=True)\n",
+    "# automatic upload of everything in /output folder doesn't work for large dataset\n",
+    "# model file has to be saved to a temp location, then uploaded by upload_file function\n",
+    "joblib.dump(value=model, filename=MODEL_FILE_NAME)\n",
+    "\n",
+    "run.upload_file(OUTPUT_FILE_NAME, MODEL_FILE_NAME)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "'./movielens-sar/reco_utils/'"
+      ]
+     },
+     "execution_count": 10,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# copy dependent python files\n",
+    "UTILS_FOLDER = './movielens-sar/reco_utils/'\n",
+    "shutil.rmtree(UTILS_FOLDER, ignore_errors=True)\n",
```

Comment:
this folder shouldn't exist already since you created `movielens-sar` in L335

Reply:
This is for running the script repeatedly

Reply:
can you add a comment then: `in case the directory already exists from a previous run`

Reply:
Also, may be we should call it `UTILS_DIR`. `Folder` is very windows like nomenclature.

Reply:
It's a good idea. Will do both comment and UTIL_DIR